students/14th/team8/minseo-kim/project_westagram/user/views.py

```diff
@@ -0,0 +1,71 @@
+import json
+import re
+from django.views import View
+from django.http import JsonResponse, HttpResponse
+from .models import User
+
+
+class SignupView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            user_data = {
+                'name': data['name'],
+                'email': data['email'],
+                'password': data['password'],
+                'number': data['number']
+            }
+
+            email_validation = re.compile(
+                r'^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+
+            if len(user_data['password']) < 8:
+                return JsonResponse(
+                    {'message':'PASSWORD_VALIDATION'}, status=400
+                )
+
+            if not re.match(email_validation, user_data['email']):
+                return JsonResponse(
+                    {'message':'EMAIL_VALIDATION'}, status=400
+                    )
+
+            if User.objects.filter(name=user_data['name']):
+                return JsonResponse(
+                    {'message':'INVALID_NAME'},status=400)
+            
+            if User.objects.filter(email=user_data['email']):
+                return JsonResponse(
+                    {'message':'INVALID_EMAIL'},status=400)
+            if User.objects.filter(number=user_data['number']):
+                return JsonResponse(
+                    {'message':'INVALID_NUMBER'}, status=400)
```

Comment:
```suggestion
                return JsonResponse({'message':'INVALID_NAME'},status=400)
            
            if User.objects.filter(email=user_data['email']):
                return JsonResponse({'message':'INVALID_EMAIL'},status=400)
                
            if User.objects.filter(number=user_data['number']):
                return JsonResponse({'message':'INVALID_NUMBER'}, status=400)
```
코드가 많이 길지 않기 때문에 한 줄에 해주셔도 좋습니다 :)

Reply:
네